### ISSUE TITLE
RPC query requests create more than 100k tcp connections 

### ISSUE THREAD

**Issue by eddietanshuo** [VictoriaMetrics/VictoriaMetrics#2508]
**Describe the bug**
version:
[v1.76.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.76.0)

vmselect -> vmstorage rpc requests create more than 100k tcp connections for a single vmstorage node.

We have 10 vmselect running with following args:
```
            - '--cacheDataPath=/cache'
            - >-
              --storageNode=victoria-metrics-victoria-metrics-cluster-vmstorage-0.victoria-metrics-victoria-metrics-cluster-vmstorage.falcon.svc.cluster.local:8401
            - >-
              --storageNode=victoria-metrics-victoria-metrics-cluster-vmstorage-1.victoria-metrics-victoria-metrics-cluster-vmstorage.falcon.svc.cluster.local:8401  
            - >-
              --storageNode=victoria-metrics-victoria-metrics-cluster-vmstorage-2.victoria-metrics-victoria-metrics-cluster-vmstorage.falcon.svc.cluster.local:8401
            - >-
              --storageNode=victoria-metrics-victoria-metrics-cluster-vmstorage-3.victoria-metrics-victoria-metrics-cluster-vmstorage.falcon.svc.cluster.local:8401
            - >-
              --storageNode=victoria-metrics-victoria-metrics-cluster-vmstorage-4.victoria-metrics-victoria-metrics-cluster-vmstorage.falcon.svc.cluster.local:8401
            - >-
              --storageNode=victoria-metrics-victoria-metrics-cluster-vmstorage-5.victoria-metrics-victoria-metrics-cluster-vmstorage.falcon.svc.cluster.local:8401
            - >-
              --storageNode=victoria-metrics-victoria-metrics-cluster-vmstorage-6.victoria-metrics-victoria-metrics-cluster-vmstorage.falcon.svc.cluster.local:8401
            - >-
              --storageNode=victoria-metrics-victoria-metrics-cluster-vmstorage-7.victoria-metrics-victoria-metrics-cluster-vmstorage.falcon.svc.cluster.local:8401
            - '--dedup.minScrapeInterval=5s'
            - '--envflag.enable=true'
            - '--envflag.prefix=VM_'
            - '--loggerFormat=json'
            - '--replicationFactor=2'
            - '--search.minStalenessInterval=5m'
```

Each vmstorage node has the same cpu/mem which is applied from aws with the same machine type.

After we restart vmstorages, we observed node-1's tcp connection reached 200k:
![image](https://user-images.githubusercontent.com/103412360/165305065-4687625b-1edc-4804-a2f2-8956394b8ee5.png)

We run `top` and found the `avg load` reached 300 for the node (64C 256G),  while the other nodes' `avg load` were around 20-50.

We use `netstat | wc -l` on the vmstorage node and found the num of connection was  210k, most of them were from clients of vmselect, and almost all of them have zero io-buffer len.

We tried to `tcpdump` some connections, we got no packet transmit, most of them are not in i/o status.

In addition, the reachability of that node is OK during that time. The overall query_range QPS was 2k at that time.
![image](https://user-images.githubusercontent.com/103412360/165306972-9e16a133-b24c-4503-b5b7-6d70aa23ebbf.png)
![image](https://user-images.githubusercontent.com/103412360/165310065-aaef6d97-e157-4f89-a38e-beb4d21b51d6.png)

We restart all vmselect, then the issues was solved:

![image](https://user-images.githubusercontent.com/103412360/165308311-44cf7d09-8411-4a26-ac79-bff9106fb315.png)

**one possibility**

we guess there could be connection pool leaking.  The vmselect could dial a lot of new connections at the moment it started serving and put all new created connections into the connection pool after they get response. 

![image](https://user-images.githubusercontent.com/103412360/165309285-49d6d30c-5de8-4def-8515-e1df7f6a0240.png)

It's also odd to get the most recent connection when we reuse a connection. (to get the last connection in the array and put it back to the array), so the older connection will never be reused or checking timeout.
![image](https://user-images.githubusercontent.com/103412360/165310817-67c6ec1e-77ae-4b3e-bc19-2d10002dc2ef.png)
![image](https://user-images.githubusercontent.com/103412360/165311131-1f35755c-e059-4a2b-8983-93eb3c5f286d.png)

**Comment by eddietanshuo** [comment]
![image](https://user-images.githubusercontent.com/103412360/165463327-2a407837-ee0d-4391-a734-201242a466be.png)

we also dump one of problem goroutine stacks, it seems goroutine has been blocked for 10 hours

**Comment by eddietanshuo** [comment]
I just downloaded the log from the problem vmstorage node after one of vmselect clients restarted:

1. First we see more 100k lines of closing infomation printed for the old vmselect(10.51.3.87).
2. Then we see there are many `accepted` and `processing` logs output for the new vmselect(10.51.8.15) a moment later.
3. At last we wait for 10 minute, no `closing` log printed 


```
{"ts":"2022-04-28T06:25:22.130Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55680"}
{"ts":"2022-04-28T06:25:22.131Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57856"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55490"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57442"}
{"ts":"2022-04-28T06:25:22.131Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57604"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57460"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55504"}
{"ts":"2022-04-28T06:25:22.131Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57616"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57468"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55508"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55532"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55500"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55510"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57464"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57480"}
{"ts":"2022-04-28T06:25:22.133Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57738"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55582"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55512"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57504"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55616"}
{"ts":"2022-04-28T06:25:22.134Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57868"}
{"ts":"2022-04-28T06:25:22.134Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57618"}
{"ts":"2022-04-28T06:25:22.134Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57626"}
{"ts":"2022-04-28T06:25:22.134Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57974"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55612"}
{"ts":"2022-04-28T06:25:22.136Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57640"}
{"ts":"2022-04-28T06:25:22.137Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57876"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57484"}
{"ts":"2022-04-28T06:25:22.137Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57878"}
{"ts":"2022-04-28T06:25:22.137Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57892"}
{"ts":"2022-04-28T06:25:22.137Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57898"}
{"ts":"2022-04-28T06:25:22.137Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57884"}
{"ts":"2022-04-28T06:25:22.137Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57900"}
{"ts":"2022-04-28T06:25:22.138Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57902"}
{"ts":"2022-04-28T06:25:22.138Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57918"}
{"ts":"2022-04-28T06:25:22.138Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57930"}
{"ts":"2022-04-28T06:25:22.138Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57920"}
{"ts":"2022-04-28T06:25:22.138Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57928"}
{"ts":"2022-04-28T06:25:22.138Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57986"}
{"ts":"2022-04-28T06:25:22.139Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57984"}
{"ts":"2022-04-28T06:25:22.134Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57980"}
{"ts":"2022-04-28T06:25:22.140Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57654"}
{"ts":"2022-04-28T06:25:22.140Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57652"}
{"ts":"2022-04-28T06:25:22.140Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57642"}
{"ts":"2022-04-28T06:25:22.140Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57664"}
{"ts":"2022-04-28T06:25:22.140Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:58006"}
{"ts":"2022-04-28T06:25:22.141Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:58014"}
{"ts":"2022-04-28T06:25:22.141Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:58008"}
{"ts":"2022-04-28T06:25:22.141Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:58010"}
{"ts":"2022-04-28T06:25:22.141Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57672"}
{"ts":"2022-04-28T06:25:22.141Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57666"}
{"ts":"2022-04-28T06:25:22.142Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57988"}
{"ts":"2022-04-28T06:25:22.142Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57682"}
{"ts":"2022-04-28T06:25:22.142Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:58018"}
{"ts":"2022-04-28T06:25:22.142Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:58016"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55542"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55540"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55622"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57502"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55548"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55550"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55626"}
{"ts":"2022-04-28T06:25:22.121Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57506"}
{"ts":"2022-04-28T06:25:22.121Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55556"}
{"ts":"2022-04-28T06:25:22.121Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55630"}
{"ts":"2022-04-28T06:25:22.121Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57524"}
{"ts":"2022-04-28T06:25:22.121Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55566"}
{"ts":"2022-04-28T06:25:22.121Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55640"}
{"ts":"2022-04-28T06:25:22.121Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55568"}
{"ts":"2022-04-28T06:25:22.121Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55656"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55618"}
{"ts":"2022-04-28T06:25:22.121Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57550"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57452"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57474"}
{"ts":"2022-04-28T06:25:22.145Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57622"}
{"ts":"2022-04-28T06:25:22.120Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:55534"}
{"ts":"2022-04-28T06:25:22.134Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57624"}
{"ts":"2022-04-28T06:25:22.146Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:58000"}
{"ts":"2022-04-28T06:25:22.130Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:57838"}
{"ts":"2022-04-28T06:25:22.146Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:202","msg":"closing vmselect conn from 10.51.3.87:58020"}
{"ts":"2022-04-28T06:25:43.237Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:164","msg":"accepted vmselect conn from 10.51.8.15:56204"}
{"ts":"2022-04-28T06:25:43.238Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:207","msg":"processing vmselect conn from 10.51.8.15:56204"}
{"ts":"2022-04-28T06:25:43.263Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:164","msg":"accepted vmselect conn from 10.51.8.15:56232"}
{"ts":"2022-04-28T06:25:43.264Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:207","msg":"processing vmselect conn from 10.51.8.15:56232"}
{"ts":"2022-04-28T06:26:02.752Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:164","msg":"accepted vmselect conn from 10.51.8.15:56500"}
{"ts":"2022-04-28T06:26:02.752Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:164","msg":"accepted vmselect conn from 10.51.8.15:56504"}
{"ts":"2022-04-28T06:26:02.757Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:207","msg":"processing vmselect conn from 10.51.8.15:56504"}
{"ts":"2022-04-28T06:26:02.758Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:207","msg":"processing vmselect conn from 10.51.8.15:56500"}
{"ts":"2022-04-28T06:26:02.891Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:164","msg":"accepted vmselect conn from 10.51.8.15:56520"}
{"ts":"2022-04-28T06:26:02.908Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:207","msg":"processing vmselect conn from 10.51.8.15:56520"}
{"ts":"2022-04-28T06:26:03.567Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:164","msg":"accepted vmselect conn from 10.51.8.15:56618"}
{"ts":"2022-04-28T06:26:03.568Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:164","msg":"accepted vmselect conn from 10.51.8.15:56616"}
{"ts":"2022-04-28T06:26:03.574Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:207","msg":"processing vmselect conn from 10.51.8.15:56618"}
{"ts":"2022-04-28T06:26:03.574Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:207","msg":"processing vmselect conn from 10.51.8.15:56616"}
{"ts":"2022-04-28T06:26:03.703Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:164","msg":"accepted vmselect conn from 10.51.8.15:56638"}
{"ts":"2022-04-28T06:26:03.703Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:164","msg":"accepted vmselect conn from 10.51.8.15:56640"}
{"ts":"2022-04-28T06:26:03.717Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:207","msg":"processing vmselect conn from 10.51.8.15:56638"}
{"ts":"2022-04-28T06:26:03.717Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:164","msg":"accepted vmselect conn from 10.51.8.15:56650"}
{"ts":"2022-04-28T06:26:03.718Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:207","msg":"processing vmselect conn from 10.51.8.15:56640"}
{"ts":"2022-04-28T06:26:03.732Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:207","msg":"processing vmselect conn from 10.51.8.15:56650"}
{"ts":"2022-04-28T06:26:02.884Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:164","msg":"accepted vmselect conn from 10.51.8.15:35730"}
{"ts":"2022-04-28T06:26:02.884Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:207","msg":"processing vmselect conn from 10.51.8.15:35730"}
{"ts":"2022-04-28T06:26:04.267Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:164","msg":"accepted vmselect conn from 10.51.8.15:35906"}
{"ts":"2022-04-28T06:26:04.269Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:207","msg":"processing vmselect conn from 10.51.8.15:35906"}
{"ts":"2022-04-28T06:26:04.345Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:164","msg":"accepted vmselect conn from 10.51.8.15:35942"}
{"ts":"2022-04-28T06:26:04.345Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:164","msg":"accepted vmselect conn from 10.51.8.15:35944"}
{"ts":"2022-04-28T06:26:04.346Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:207","msg":"processing vmselect conn from 10.51.8.15:35944"}
{"ts":"2022-04-28T06:26:04.348Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:207","msg":"processing vmselect conn from 10.51.8.15:35942"}
{"ts":"2022-04-28T06:26:04.430Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:164","msg":"accepted vmselect conn from 10.51.8.15:36016"}
{"ts":"2022-04-28T06:26:04.439Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:164","msg":"accepted vmselect conn from 10.51.8.15:36018"}
{"ts":"2022-04-28T06:26:04.439Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:207","msg":"processing vmselect conn from 10.51.8.15:36018"}
{"ts":"2022-04-28T06:26:04.440Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:207","msg":"processing vmselect conn from 10.51.8.15:36016"}
{"ts":"2022-04-28T06:29:41.038Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1264","msg":"merged 58941573 rows across 52954010 blocks in 41.267 seconds at 1428282 rows/sec to \"/storage/data/small/2022_04/58759047_26060024_20220421052700.000_20220428063620.000_16E9AA2855AF150E\"; sizeBytes: 755900480"}
{"ts":"2022-04-28T06:35:35.497Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1264","msg":"merged 58070765 rows across 52787028 blocks in 39.035 seconds at 1487675 rows/sec to \"/storage/data/small/2022_04/57840668_26187744_20220421055100.000_20220428064220.000_16E9AA2855AF1556\"; sizeBytes: 726979698"}
{"ts":"2022-04-28T06:39:04.147Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:164","msg":"accepted vmselect conn from 10.51.8.15:44244"}
{"ts":"2022-04-28T06:39:04.147Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:164","msg":"accepted vmselect conn from 10.51.8.15:44246"}
{"ts":"2022-04-28T06:39:04.154Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:207","msg":"processing vmselect conn from 10.51.8.15:44244"}
{"ts":"2022-04-28T06:39:04.158Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/transport/server.go:207","msg":"processing vmselect conn from 10.51.8.15:44246"}
{"ts":"2022-04-28T06:44:02.493Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1264","msg":"merged 102674730 rows across 87297703 blocks in 66.891 seconds at 1534964 rows/sec to \"/storage/data/small/2022_04/102334777_32787682_20220421052700.000_20220428065020.000_16E9AA2855AF15B9\"; sizeBytes: 1119742647"}
{"ts":"2022-04-28T06:44:17.569Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1264","msg":"merged 5174271854 rows across 467171080 blocks in 1240.701 seconds at 4170440 rows/sec to \"/storage/data/big/2022_04/5171341813_49084914_20220420160600.000_20220428062748.000_16E9AA2855AF14CB\"; sizeBytes: 16946284309"}

```

**Comment by eddietanshuo** [comment]
@valyala Thanks for update. For some reason I failed to build the version at commit https://github.com/VictoriaMetrics/VictoriaMetrics/commit/190c8b463c206d04d731c1317ca25a833ae2482d . When I run the make package, it generated `error obtaining VCS status: exit status 128`

I replaced the whole `conn_pool.go` file with your fix. It seems the fix **SOLVE** my problem. It's very useful. I also make a snapshot for evidence:
![image](https://user-images.githubusercontent.com/103412360/166659776-6191b850-7a1d-4150-9865-1246dd9dc627.png)

We still need to take more time to run it on our enviroment to test stability. Thanks again for help

**Comment by valyala** [comment]
@eddietanshuo , thanks for the confirmation!

The bugfix, which closes `vmselect->vmstorage` connections, which were idle for more than 30 seconds, has been included in [VictoriaMetrics v1.77.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed then.
